torch/nn/parallel/distributed.py

```diff
@@ -654,6 +657,30 @@ def _register_comm_hook(self, state: object, hook: callable):
             >>>     fut.set_result(bucket.get_tensors())
             >>>     return fut
 
+        Example::
+            Below is an example of a simple allreduce hook.
+
+            >>> ddp._register_comm_hook(state = None, hook = allreduce)
```

Comment:
don't you need to flip the order of this the allreduce below, otherwise this would throw "NameError: name 'allreduce' is not defined"